### ISSUE TITLE
Resx stuck at "Opening files..." when on VPN connection

### ISSUE THREAD

**Issue by natidea** [dotnet/project-system#2822]
From [VSFeedback](https://developercommunity.visualstudio.com/content/problem/67262/resx-stuck-at-opening-files.html) | [448768](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/448768)

> Either by dbl-clicking it, or opening the solution with it opened, VS gets stucked at "Opening the files..." and won't close that popup.
> 
> I've waited several minutes to no avail.
> 
> Thing is, and I find this very strange, it only works if I disconnect from the VPN. Yes, you read that right.
> 
> On the work Network, no problems, the file open.
> On the VPN, stuck at "Opening the file..."
> 

The issue appeared to go away on a 15.3 Preview build, but is still present in the released bits.

More discussion on the [feedback ticket](https://developercommunity.visualstudio.com/content/problem/67262/resx-stuck-at-opening-files.html).

**Comment by jspetit** [comment]
Thanks for opening that issue Nat.

I'd love to try to identify the cause of my problem, so could someone give me a hint as to where to start looking?

**Comment by jspetit** [comment]
Let me summarize the problem:

The problem occurs when :

1. I'm connected to our Company VPN;
2. I attempt to open a .resx file, either
    a) by opening directly in Solution explorer
    b) by opening my project while the file was opened the last time VS was closed
3. The popup "Opening the files..." is displayed
4. The editor window never opens and the popup remain indefinitely open and VS is unusable in the meantime 
5.  As soon as I logoff the VPN, the editor window displays.

I know my colleagues had the same issue, but Nat didn't, which raise the possibility that VS is trying to access a site that is locked by our VPN.

So, is it a Visual Studio bug, or a project-system issue?

**Comment by Pilchie** [comment]
One thing that might help is a dump file:

* Start the 32-bit Task Manager from `C:\Windows\SysWOW64\taskmgr.exe`
* Switch to the processes tab
* Right click on `devenv.exe` and choose "Create dump file..."
* Upload the (very large) dump file somewhere for us to take a look.

**Comment by jspetit** [comment]
![image](https://user-images.githubusercontent.com/5932705/30444669-b865c9da-9951-11e7-9290-ef6d858c0202.png)

I'll upload the dump file shortly. Here is the printscreen of the opening the file popup.

**Comment by jspetit** [comment]
An observation: 

I was able to open a resx in one or my .Net Library, but could not open the ones in my web site project, a C# Web Application:

`    <ProjectTypeGuids>{349c5851-65df-11da-9384-00065b846f21};{fae04ec0-301f-11d3-bf4b-00c04f79efbc}</ProjectTypeGuids>
`
Don't know if it helps...

**Comment by Pilchie** [comment]
@Tanya-Solyanik the dump file shows the UI thread in a CreateFile call in the stack below, with a file in the obj directory `...obj\x86\Debug\TempPE\Areas.Admin.Resources.Views.ClientStats.Designer.cs.dll`
Any idea how a VPN could affect this?

@jspetit any chance it could be AV software or something instead of VPN?

```
 	ntdll.dll!_ZwCreateFile@44 Line 783	Unknown	Symbols loaded.
 	KERNELBASE.dll!CreateFileW Line 575	C	Symbols loaded.
>	kernel32.dll!CreateFileWImplementation Line 237	C	Symbols loaded.
 	[Managed to Native Transition]		Annotated Frame
 	mscorlib.dll!System.Reflection.AssemblyName.GetAssemblyName Line 213	C#	Symbols loaded.
 	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSTypeResolutionService.AssemblyEntry.AssemblyName.get Line 3881	C#	Symbols loaded.
 	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSTypeResolutionService.AssemblyEntry.FullName.get Line 3902	C#	Symbols loaded.
 	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSTypeResolutionService.FindAssembly Line 3228	C#	Symbols loaded.
 	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSTypeResolutionService.SearchGeneratedEntries Line 3288	C#	Symbols loaded.
 	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSTypeResolutionService.System.ComponentModel.Design.IDesignTimeAssemblyLoader.GetTargetAssemblyPath Line 3364	C#	Symbols loaded.
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssembly Line 610	C#	Symbols loaded.
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetTypeFromTargetLocation Line 356	C#	Symbols loaded.
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.IsSupportedType Line 489	C#	Symbols loaded.
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkProvider.IsSupportedType Line 238	C#	Symbols loaded.
 	Microsoft.VisualStudio.DataDesign.Common.dll!Microsoft.VSDesigner.MultiTargetService.IsSupportedType Line 194	C#	Symbols loaded.
 	Microsoft.VisualStudio.DataDesign.Common.dll!Microsoft.VSDesigner.MultiTargetService.GetSupportedType Line 272	C#	Symbols loaded.
 	Microsoft.VisualStudio.DataDesign.Common.dll!Microsoft.VSDesigner.MultiTargetService.TypeNameConverter Line 224	C#	Symbols loaded.
 	Microsoft.VisualStudio.Editors.dll!Microsoft.VisualStudio.Editors.ResourceEditor.ResourceFile.TypeNameConverter Line 877	Basic	Symbols loaded.
 	Microsoft.VisualStudio.Editors.dll!Microsoft.VisualStudio.Editors.ResourceEditor.Resource.TypeNameConverter Line 462	Basic	Symbols loaded.
 	System.Windows.Forms.dll!System.Resources.MultitargetUtil.GetAssemblyQualifiedName Line 30	C#	Symbols loaded.
 	System.Windows.Forms.dll!System.Resources.ResXDataNode.GetValueTypeName Line 598	C#	Symbols loaded.
 	Microsoft.VisualStudio.Editors.dll!Microsoft.VisualStudio.Editors.ResourceEditor.Resource.get_ValueTypeName Line 1155	Basic	Symbols loaded.
 	Microsoft.VisualStudio.Editors.dll!Microsoft.VisualStudio.Editors.ResourceEditor.Resource.CheckValueForErrors Line 2072	Basic	Symbols loaded.
 	Microsoft.VisualStudio.Editors.dll!Microsoft.VisualStudio.Editors.ResourceEditor.ResourceFile.ComponentChangeService_ComponentAdded Line 601	Basic	Symbols loaded.
```

**Comment by jspetit** [comment]
Doubtful, my McAfee AV is on all the times, and the problem only occurs while on VPN.

The only (far fetch) possibility on this side would be Mcafee blocking the file while attempting to connect to another server to validate the file. 

Like I said, doubtful...

**Comment by jspetit** [comment]
Where does the info used to "Create" the file in OBJ come from? Is the template on the web somehow?

Even when not on the VPN, opening a RESX for the first time in the session takes a few seconds (longer than I would expect). 

Could VS be getting a template from the web, then use the "cached" info, only to loose the cache on exit?

**Comment by davkean** [comment]
An ETL trace would help us get to the bottom where the cost is here, can you do:

1. Download PerfView from http://www.microsoft.com/en-us/download/details.aspx?id=28567 and copy to a temp directory.
2. From an admin command prompt, run the following to start recording:
perfview.exe collect c:\temp\ReproTrace.etl -CircularMB:2048 -Merge:true -Zip:true -Providers:641d7f6c-481c-42e8-ab7e-d18dc5e5cb9e,*Microsoft-VisualStudio-Common -ThreadTime -NoV2Rundown
This will launch PerfView UI and start recording automatically. After 3-5 seconds, you can start Visual Studio to reproduce the issue.
3. Once the issue is reproduced, you can click “Stop Collection” on the PerfView UI. It will take a while to process the trace, but at the end there should be a zipped file at c:\temp\ReproTrace.etl.zip.
4. Trace will have up to several minutes of information only so it is important to stop recording as soon as the issue is reproduced.

**Comment by jspetit** [comment]
This link gives a 404...

Found it at
https://www.microsoft.com/en-ca/download/details.aspx?id=28567

Maybe because I'm in Canada?

________________________________

**Comment by jspetit** [comment]
Here is the link to the ETL trace file:
[https://1drv.ms/u/s!Ah6dyWzRZdREnwWs_3it4yKqbyq-](url)

However, the behavior of VS this time was different.

Prior to starting the VPN and the trace, I loaded my project and made sure that the resource file was opened, then closed VS.

I started the trace, and 30 seconds later, I opened the project directly, which opened VS.

However, the message "Loading Solution" stayed there for several minutes, not "opening the file", and when VS finally opened, the ressource file opened as well.

But it was painfully slow.

Maybe it was because I pre-loaded the resource file beforehand, or the bug is intermittant.

I'm willing to retry tomorrow, if you don't find anything. I can reboot, and start the vpn, perfview and VS without "Warming up" the resource editor...

Good hunting!

**Comment by jspetit** [comment]
Here is another ETL trace file: 
[https://1drv.ms/u/s!Ah6dyWzRZdREnwZTkf_yjZlxnxC9](url)

This time I simply booted, started the vpn, VS, ETL Trace and opened a RESX.

I stopped it 6 minutes later, while the opening the file popup was still on and VS greyed out (not responding).

So I started it at roughly 8:35 and stopped collecting at 8:41ish. And at 9:05 am it was still "opening"

It turns out that my resx eventually opened., as at 9:30 it was there. I wasn't at my desk so I don,t know exactly when it opened, but you agree with me that it sits between 30-50 minutes.

I hope you find something!

**Comment by jspetit** [comment]
Hello, just dropping by to see the progress of this bug....

**Comment by jspetit** [comment]
Well, I don't know what you did, but it's fixed in 15.5.1. Unless something changed on our VPN?

**Comment by Pilchie** [comment]
I don't think we did anything, but I'm going to close this, as we don't have a repro for this anymore :(
